### ISSUE TITLE
Crash after optimization is done

### ISSUE THREAD

**Issue by application-developer-DA** [shssoichiro/oxipng#89]
```
Processing: result.png
    16x16 pixels, PNG format
    4x8 bits/pixel, RGBA
    IDAT size = 63 bytes
    File size = 120 bytes
Reducing image to 1x1 bits/pixel, Grayscale
Trying: 8 combinations
Found better combination:
    zc = 9  zm = 9  zs = 0  f = 0        27 bytes
    IDAT size = 27 bytes (36 bytes decrease)
    file size = 84 bytes (36 bytes = 30.00% decrease)
The resulting image is corrupted and will not be outputted.
This is a bug! Please report it at https://github.com/shssoichiro/oxipng/issues
thread 'main' panicked at 'end', src/main.rs:78:4
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

The test image is attached. Let me know if you need more information.

![result](https://user-images.githubusercontent.com/2765698/31446704-7280bb0a-aea0-11e7-9c62-8ef75384497e.png)

**Comment by shssoichiro** [comment]
Thanks for reporting this! I will take a look and get this fixed soon.

**Comment by shssoichiro** [comment]
Fixed in 0.18.1.
